pkg/beholder/client.go

```diff
@@ -269,17 +269,20 @@ func newTracerProvider(config Config, resource *sdkresource.Resource, creds cred
 	if err != nil {
 		return nil, err
 	}
-	tp := sdktrace.NewTracerProvider(
-		sdktrace.WithBatcher(exporter,
-			trace.WithBatchTimeout(config.TraceBatchTimeout)), // Default is 5s
+
+	opts := []sdktrace.TracerProviderOption{
+		sdktrace.WithBatcher(exporter, trace.WithBatchTimeout(config.TraceBatchTimeout)), // Default is 5s
 		sdktrace.WithResource(resource),
 		sdktrace.WithSampler(
 			sdktrace.ParentBased(
 				sdktrace.TraceIDRatioBased(config.TraceSampleRatio),
 			),
 		),
-	)
-	return tp, nil
+	}
+	if config.TraceSpanExporter != nil {
+		opts = append(opts, sdktrace.WithBatcher(config.TraceSpanExporter))
```

Comment:
@pkcll @patrickhuie19 Does this optional config for an additional exporter make sense? Or should we consider other alternatives?

Reply:
Passing the configured exporter is more of a dependency injection rather than part of the config. Passing an exporter as part of the config seems like mixing different types of code aspects.

Reply:
I see 4 options:
- loop.telem.SetupTracing calling beholder.newTraceProvider
- beholder.newTraceProvider calling loop.telem.SetupTracing
- loop.telem.SetupTracing and beholder.newTraceProvider calling some "common denominator" function e.g telemetry.NewTraceProvider
- both loop.telem.SetupTracing and beholder.newTraceProvider using shared configuration

Reply:
What does the TraceSpanExporter do on top of everything else configured in `opts`?

Reply:
This is one approach to bridging the two tracing systems.

Reply:
> Passing the configured exporter is more of a dependency injection rather than part of the config. Passing an exporter as part of the config seems like mixing different types of code aspects.

Otel actually use config as an "umbrella" struct for both pure configuration (collection of option fields) and dependencies/sub-components 

See here: https://github.com/open-telemetry/opentelemetry-go/blob/fe6c67e7e9b408d7f1e05356e4ccafadf0475b10/sdk/trace/provider.go#L26-L45

```
type tracerProviderConfig struct {
	processors []SpanProcessor
	sampler Sampler
	idGenerator IDGenerator
	spanLimits SpanLimits
	resource *resource.Resource
}
```

The question is do we want to follow chainlink/chainlink-common config semantics or use it the way Otel does it.

Reply:
> What does the TraceSpanExporter do on top of everything else configured in opts?  
@patrickhuie19 
It allows to override the default otel.Exporter configured in in NewOtelClient/newTracerProvider

Reply:
It adds an additional exporter. It does not replace the normal one. They are accumulated: https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/trace/provider.go#L338

Reply:
You are right. 
Do you see use case for having two separate trace processors/exporters for loop/server otel instrumentation and for custom traces produced from code?

Reply:
@jmank88 @patrickhuie19, ^^^ WDYT

Reply:
We could add a check for duplicate endpoints. But otherwise using both at the same time is not inherently problematic, and having the flexibility will make it easier to transition as well as support non-production cases.

Reply:
> having the flexibility will make it easier to transition 

Makes sense